Responsive Boat Theme PR

Alexander-Torres opened the pull request:
I made a mistake and merged my developer branch with my master. But here's what I have done with my theme so far. The files can be found in **modules > apps > foundation > frontend-theme > responsive-boat-theme**.

I kept all css in the _custom.css file to make it easier to read all of it at once. I also kept all JS in the main.js file, but spread some changes among portal-normal.ftl, navigation.ftl, and portlet.ftl.
>   **Portal-Normal:** Switched navigation and header section positions to make it easier to keep the header below the navigation bar when it switches from being positionally-fixed to static. Also added script tag to link main.js.
>   **Navigation:** Replaced hard-coded navigation with a dynamically imported navigation portlet, and added a button.
>   **Portlet:** Changed portlet title text from h2 to h1 element so that it makes more sense giving it the largest font in the Portlet.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+/* inject:imports */
+/* endinject */
+
+/* This file allows you to override default styles in one central location for easier upgrade and maintenance. */
```
Remove unused CSS comments.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+/* This file allows you to override default styles in one central location for easier upgrade and maintenance. */
+
+
+/********** VARIABLE SECTION **********/
```
CSS comments should use a standard format.

Ex.
```
/* ---------- Variables ---------- */
```

jonmak08 replied:
We do not need a comment around the variable since we can already tell that they are SASS variables.

jonmak08 replied:
Line break needed below CSS comment.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+$expandTime: 0.4s; 				// time it takes for mobile navbar to expand or collapse
+$fadeTime: 0.5s; 				// time it takes for navbar links to change color when hovered over
+$navBtnPad: 20px; 				// padding above and below nav-button
+$navBtnHeight: 40px; 			// height of nav-button
```
Do we really need comments for these variables?  The variable name should be detailed enough to provide this information.  If it is not, we should rename the SASS variable.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+
+
+/********** VARIABLE SECTION **********/
+$expandTime: 0.4s; 				// time it takes for mobile navbar to expand or collapse
```
Since this is relating to mobile, I would rename this variable to be a little more descriptive about what it does rather than relying on the comments for further explanation.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+/********** VARIABLE SECTION **********/
+$expandTime: 0.4s; 				// time it takes for mobile navbar to expand or collapse
+$fadeTime: 0.5s; 				// time it takes for navbar links to change color when hovered over
+$navBtnPad: 20px; 				// padding above and below nav-button
```
We should rename this to spell out ```Padding``` instead of shortening it.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+$fadeTime: 0.5s; 				// time it takes for navbar links to change color when hovered over
+$navBtnPad: 20px; 				// padding above and below nav-button
+$navBtnHeight: 40px; 			// height of nav-button
+/********** END OF VARIABLE SECTION **********/
```
We do not need an ending comment.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+/********** END OF VARIABLE SECTION **********/
+
+
+
```
There should only be 1 line break between CSS for spacing.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+		size: cover;
+		position-x: center;
+		attachment: fixed; 		// background stays fixed while scrolling
+	}
```
Alphabetize CSS rules.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+		image: url("../images/background.jpg");
+		size: cover;
+		position-x: center;
+		attachment: fixed; 		// background stays fixed while scrolling
```
We should spell out each CSS rule or short hand it instead of defining everything within this block.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+
+	#banner {
+		padding-bottom: 500px;	// add space below nave to copy the Wordpress theme
+		color: white;
```
Colors should be written as HEX values whenever possible.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+	#banner {
+		padding-bottom: 500px;	// add space below nave to copy the Wordpress theme
+		color: white;
+		background: rgba(0,0,0,0.5);
```
Space missing between comma separated values.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+		width: 100%;
+		z-index: 1; 			// found that some portlet content overlaps the navigation bar unless z-index is set to 1 or higher
+
+		div.nav-btn { display: none; } // hide mobile nav-button until screen width is small enough
```
This needs to be formatted correctly.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+		div.nav-btn { display: none; } // hide mobile nav-button until screen width is small enough
+
+		div, ul, .navbar, .portlet {
+			border: none; 		// eliminate unnecessary borders that just make navbar taller (that were set outside of my css)
```
Instead of ```border: none;``` we should write it as ```border-width: 0````.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+			top: unset;
+			bottom: 0;
+			border-radius: 4px;
+			width: 200px;
```
Line break needed to space out CSS selectors.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+			}
+		}
+
+		.portlet:hover > .portlet-topper, .portlet.focus > .portlet-topper {
```
We should be able to SASS this.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+			transform: translateY(100%); // change direction that portlet topper floats
+		}
+
+		.navbar, .portlet, ul { margin: 0; } // eliminate unnecessary and pesky margins in navbar (also set outside of my css)
```
Fix formatting.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+					transition: color $fadeTime;
+				}
+
+				a:hover {
```
We should be able to SASS this.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/css/_custom.scss:
```diff
@@ -0,0 +1,278 @@
+			}
+		}
+
+		@media (max-width: 770px) { // change navbar look for mobile devices
```
Can we use Liferay's default media queries instead of defining our own?

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/js/main.js:
```diff
@@ -0,0 +1,52 @@
+/* This function gets loaded when all the HTML, not including the portlets, is loaded. */
+AUI().ready(function() {}); /******************** NOT USED ********************/
```
Remove unused JS.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/js/main.js:
```diff
@@ -0,0 +1,52 @@
+function myPageJS() {
+	let navBar = document.getElementById('navigation'),
+		navBtn = document.querySelector('#navigation div.nav-btn'),
+		portletPs = document.querySelectorAll('#content .portlet-body p');
```
Why use native JS instead of jQuery/AUI that is installed with Liferay?

jonmak08 replied:
We can rename JS variables to be more descriptive.

```Paragraphs = ...````

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/js/main.js:
```diff
@@ -0,0 +1,52 @@
+
+
+
+	/********** SCROLL EVENT HANDLER **********/
```
You should only need single line comments for JS.

jonmak08 commented on modules/apps/foundation/frontend-theme/responsive-boat-theme/src/js/main.js:
```diff
@@ -0,0 +1,52 @@
+
+// Function Containing All of My Event Handlers
+function myPageJS() {
+	let navBar = document.getElementById('navigation'),
```
We should rename the JS variables to match the node that it is selecting.

Ex.
```
navigation.
```

jonmak08 replied:
Each JS variable should be defined on its own line with their own signal.

jonmak08 commented:
Hey @Alexander-Torres,

It looks like that merge commit caused a lot of issues for you.  You have a lot of unresolved changes in your files.  You might want to revert your HEAD to before the merge commits and try again if you are trying to merge everything.

It seems like you have a lot of unnecessary files that you committed in your theme as well.  You should only commit the files that you change and are needed.  Everything else should be default and picked up on deploy of the theme in Liferay.

Please let me know if you have any questions. Thanks!

jonmak08 commented:
Hey @Alexander-Torres,

I made comments in your files, but I did not highlight all the areas that need to be updated.  You will need to apply your changes throughout your files and resend me a new pull request.

Please let me know if you have any questions. Thanks!

Alexander-Torres commented:
Hey Jonathan,

I was trying a bunch of things to fix the issues caused by the merge commit
but made a bit of a mess. So I deleted my old repository and source code,
and then re-forked the liferay repository and re-cloned the source code.
I'm doing some more training on git so that this doesn't happen again or so
that I can fix it next time.

To answer some of your questions:

   - I used native JS because it's more familiar and I like to challenge
   myself to make things work without jQuery. But since it's installed
   anyways, I replaced a lot of the code with jQuery and found that it's a bit
   simpler.


   - I didn't think of using Liferay's default media queries, so I went
   ahead and swapped mine with those.

So overall, I only updated the main.js and _custom.css files. The pull
request should be sent to you already.
